services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/functionaltests/UpdateBehaviorTest.java

```diff
@@ -145,6 +162,355 @@ public void updateBehaviors_transactWriteItems_secondUpdate() {
         assertThat(persistedRecord.getCreatedAutoUpdateOn()).isEqualTo(firstUpdatedRecord.getCreatedAutoUpdateOn());
     }
 
+    @Test
+    public void when_updatingNestedObjectWithSingleLevel_existingInformationIsPreserved_scalar_only_update() {
+
+        NestedRecordWithUpdateBehavior nestedRecord = createNestedWithDefaults("id456", 5L);
+
+        RecordWithUpdateBehaviors record = new RecordWithUpdateBehaviors();
+        record.setId("id123");
+        record.setNestedRecord(nestedRecord);
+
+        mappedTable.putItem(record);
+
+        NestedRecordWithUpdateBehavior updatedNestedRecord = new NestedRecordWithUpdateBehavior();
+        long updatedNestedCounter = 10L;
+        updatedNestedRecord.setNestedCounter(updatedNestedCounter);
+
+        RecordWithUpdateBehaviors update_record = new RecordWithUpdateBehaviors();
+        update_record.setId("id123");
+        update_record.setVersion(1L);
+        update_record.setNestedRecord(updatedNestedRecord);
+
+        mappedTable.updateItem(r -> r.item(update_record).ignoreNullsMode(IgnoreNullsMode.SCALAR_ONLY));
+
+        RecordWithUpdateBehaviors persistedRecord = mappedTable.getItem(r -> r.key(k -> k.partitionValue("id123")));
+
+        verifySingleLevelNestingTargetedUpdateBehavior(persistedRecord.getNestedRecord(), updatedNestedCounter);
+    }
+
+    @Test
+    public void when_updatingNestedObjectWithSingleLevel_default_mode_update_existingInformationIsErased() {
+
+        NestedRecordWithUpdateBehavior nestedRecord = createNestedWithDefaults("id456", 5L);
+
+        RecordWithUpdateBehaviors record = new RecordWithUpdateBehaviors();
+        record.setId("id123");
+        record.setNestedRecord(nestedRecord);
+
+        mappedTable.putItem(record);
+
+        NestedRecordWithUpdateBehavior updatedNestedRecord = new NestedRecordWithUpdateBehavior();
+        long updatedNestedCounter = 10L;
+        updatedNestedRecord.setNestedCounter(updatedNestedCounter);
+
+        RecordWithUpdateBehaviors update_record = new RecordWithUpdateBehaviors();
+        update_record.setId("id123");
+        update_record.setVersion(1L);
+        update_record.setNestedRecord(updatedNestedRecord);
+
+        mappedTable.updateItem(r -> r.item(update_record).ignoreNullsMode(IgnoreNullsMode.DEFAULT));
+
+        RecordWithUpdateBehaviors persistedRecord = mappedTable.getItem(r -> r.key(k -> k.partitionValue("id123")));
+
+        verifySingleLevelNestingDefaultUpdateBehavior(persistedRecord.getNestedRecord(), updatedNestedCounter);
+    }
+
+    @Test
```

Comment:
Can we have parameterized test so that,s it easy to find  expected behaviours

```java
public static List<Arguments> getDifferentCombinationOfignoreNullsModeAndExpectedResult() {
        return Arrays.asList(
            Arguments.of(simpleNestedAttrib, enum1, expectedMap1),
  Arguments.of(simpleNestedAttrib, enum2, expectedMap1),
  Arguments.of(nestedAttribWithEmpty, enum2, expectedMap1),
  
  etc etc


@ParameterizedTest
@MethodSource("getDifferentCombinationOfignoreNullsModeAndExpectedResult")
void testUpdateWithDifferentModes(NestedAttibute attrib,  IgnoreNullsMode ignoreNullsMode, Map<String, AttributeVale> expectedUpdate)
{
//existing common code
        mappedTable.updateItem(r -> r.item(attrib).ignoreNullsMode(ignoreNullsMode));

Map<String, AttributeVale> valueInDDB = nativeClient.get(item);

assertThat(expectedUpdate).isEqualTo(valueInDDB)

}
```

Reply:
This is not going to make it neater. For each case, i.e. simple nested attribute, empty attribute or a multi nested attribute, the validation functions are different. If at all we group some methods together, it will not make it neater.

Reply:
To make it more readable, I have grouped some of the validation methods together in the next revision into : 

- verifyMultipleLevelNestingTargetedUpdateBehavior
- verifySingleLevelNestingTargetedUpdateBehavior